tests/test_lattice.py

```diff
@@ -0,0 +1,1540 @@
+        initial_lattice.remove_sites(identifiers=["A"])
+
+        # Assert 2: The neighbor map should be empty again
+        assert not initial_lattice._neighbor_maps
+
+    def test_modification_clears_distance_matrix_cache(self, initial_lattice):
+        """
+        Tests that add_sites and remove_sites correctly invalidate the
+        cached distance matrix and that the recomputed matrix is correct.
+        """
+        # Arrange 1: Compute, cache, and perform a meaningful check on the original matrix.
+        lat = initial_lattice
+        original_matrix = lat.distance_matrix
+        assert lat._distance_matrix is not None
+        assert original_matrix.shape == (3, 3)
+        # Meaningful check: distance from 'A'(idx 0) to 'B'(idx 1) should be 1.0
+        np.testing.assert_allclose(original_matrix[0, 1], 1.0)
+
+        # Act 1: Add a site. This should invalidate the cache.
+        lat.add_sites(identifiers=["D"], coordinates=[[1, 1]])
+
+        # Assert 1: Check cache is cleared and the new matrix is correct.
+        assert lat._distance_matrix is None  # Verify cache invalidation
+        new_matrix_added = lat.distance_matrix
+        assert new_matrix_added.shape == (4, 4)
+        # Meaningful check: distance from 'B'(idx 1) to new site 'D'(idx 3) should be 1.0
+        # Coords: B=[1,0], D=[1,1]
+        np.testing.assert_allclose(new_matrix_added[1, 3], 1.0)
+
+        # Act 2: Remove a site. This should also invalidate the cache.
+        lat.remove_sites(identifiers=["A"])
+
+        # Assert 2: Check cache is cleared again and the final matrix is correct.
+        assert lat._distance_matrix is None  # Verify cache invalidation
+        final_matrix = lat.distance_matrix
+        assert final_matrix.shape == (3, 3)  # Now has 3 sites again
+        # Meaningful check: After removing 'A', the sites are B, C, D.
+        # 'B' is now at index 0 (coords [1,0])
+        # 'C' is now at index 1 (coords [0,1])
+        # 'D' is now at index 2 (coords [1,1])
+        # Distance from new 'B' (idx 0) to new 'D' (idx 2) should be 1.0
+        np.testing.assert_allclose(final_matrix[0, 2], 1.0)
+
+    def test_neighbor_finding_returns_sorted_list(self, simple_square_lattice):
+        """
+        Ensures that the list of neighbors returned by get_neighbors is always sorted.
+        This provides a stricter check than set-based comparisons.
+        """
+        # Arrange
+        lattice = simple_square_lattice
+
+        # Act
+        # Get neighbors for the central site (index 1 in a 2x2 grid)
+        # Expected neighbors are 0, 3.
+        neighbors = lattice.get_neighbors(1, k=1)
+
+        # Assert
+        # We compare directly against a pre-sorted list, not a set.
+        # This will fail if the implementation returns [3, 0] instead of [0, 3].
+        assert neighbors == [
+            0,
+            3,
+        ], "The neighbor list should be sorted in ascending order."
+
+
+class TestDistanceMatrixStrength:
```

Comment:
the name convention with Strength is weird, why not only TestDistanceMatrix or something alike?